tests/test_encoding.py

```diff
@@ -11,6 +11,20 @@
 from .test_helpers import ENCODING_FACTORIES, MAX_EXAMPLES
 
 
+@pytest.mark.skip(reason="Takes a really long time to finish, but was added to reproduce a crash.")
+@pytest.mark.parametrize("make_enc", ENCODING_FACTORIES)
+def test_extremely_big_encoding(make_enc: Callable[[], tiktoken.Encoding]):
+    enc = make_enc()
+    for c in ["^", "0", "a", " ", "\n", "'s"]:
```

Comment:
all of these are stressing different parts of the regex, makin sure none have catastrophic backtracking